test/mlir/conversion/onnx_to_mhlo/Math/Reduction.mlir

```diff
@@ -1,4 +1,4 @@
-// RUN: onnx-mlir-opt --convert-onnx-to-mhlo %s -split-input-file | FileCheck %s
+// RUN: onnx-mlir-opt --decompose-onnx="target=mhlo" --shape-inference --convert-onnx-to-mhlo %s --canonicalize -split-input-file | FileCheck %s
```

Comment:
We decompose the softmax operator into primitive operators when the target is mhlo. We add the other two options to do shape inference and canonicalize the result

Reply:
I know we may not have set the best example, the preferred way is to test only one pass at a time, so that potential errors are easier to detect. I would then do 2 test files, one to test the first pass and one to do the second pass. 

We have often added `canonicalize` as a secondary pass as often the output without canonicalize is very redundant, so using this as a second pass is fine if it makes the test easier.